diff --git a/ext/redhat/client.init b/ext/redhat/client.init
index 4ee7a2f..1a23190 100644
--- a/ext/redhat/client.init
+++ b/ext/redhat/client.init
@@ -17,7 +17,7 @@
 lockfile=/var/lock/subsys/puppet
 piddir=/var/run/puppetlabs
 pidfile="${piddir}/agent.pid"
-puppetd=/opt/puppetlabs/puppet/bin/puppet
+puppetd=/usr/bin/puppet
 pid=$(cat "$pidfile" 2> /dev/null)
 RETVAL=0

diff --git a/ext/systemd/puppet.service b/ext/systemd/puppet.service
index 7c16615..c63627d 100644
--- a/ext/systemd/puppet.service
+++ b/ext/systemd/puppet.service
@@ -11,13 +11,13 @@
 [Unit]
 Description=Puppet agent
 Wants=basic.target
-After=basic.target network.target
+After=basic.target network.target puppetmaster.service

 [Service]
 EnvironmentFile=-/etc/sysconfig/puppetagent
 EnvironmentFile=-/etc/sysconfig/puppet
 EnvironmentFile=-/etc/default/puppet
-ExecStart=/opt/puppetlabs/puppet/bin/puppet agent $PUPPET_EXTRA_OPTS --no-daemonize
+ExecStart=/usr/bin/puppet agent $PUPPET_EXTRA_OPTS --no-daemonize
 ExecReload=/bin/kill -HUP $MAINPID
 KillMode=process

diff --git a/install.rb b/install.rb
old mode 100755
new mode 100644
index 632052e..96c02dc
--- a/install.rb
+++ b/install.rb
@@ -169,13 +169,13 @@ def prepare_installation
     opts.on('--destdir[=OPTIONAL]', 'Installation prefix for all targets', 'Default essentially /') do |destdir|
       InstallOptions.destdir = destdir
     end
-    opts.on('--configdir[=OPTIONAL]', 'Installation directory for config files', 'Default /etc/puppetlabs/puppet') do |configdir|
+    opts.on('--configdir[=OPTIONAL]', 'Installation directory for config files', 'Default /etc/puppetlabs/puppet') do |destdir|
       InstallOptions.configdir = configdir
     end
     opts.on('--codedir[=OPTIONAL]', 'Installation directory for code files', 'Default /etc/puppetlabs/code') do |codedir|
       InstallOptions.codedir = codedir
     end
-    opts.on('--vardir[=OPTIONAL]', 'Installation directory for var files', 'Default /opt/puppetlabs/puppet/cache') do |vardir|
+    opts.on('--vardir[=OPTIONAL]', 'Installation directory for var files', 'Default /var/lib/puppet') do |vardir|
       InstallOptions.vardir = vardir
     end
     opts.on('--rundir[=OPTIONAL]', 'Installation directory for state files', 'Default /var/run/puppetlabs') do |rundir|
@@ -187,7 +187,7 @@ def prepare_installation
     opts.on('--bindir[=OPTIONAL]', 'Installation directory for binaries', 'overrides RbConfig::CONFIG["bindir"]') do |bindir|
       InstallOptions.bindir = bindir
     end
-    opts.on('--localedir[=OPTIONAL]', 'Installation directory for locale information', 'Default /opt/puppetlabs/puppet/share/locale') do |localedir|
+    opts.on('--localedir[=OPTIONAL]', 'Installation directory for locale information', 'Default /usr/share/puppetlabs/puppet/locale') do |localedir|
       InstallOptions.localedir = localedir
     end
     opts.on('--ruby[=OPTIONAL]', 'Ruby interpreter to use with installation', 'overrides ruby used to call install.rb') do |ruby|
@@ -273,7 +273,7 @@ def prepare_installation
   elsif $operatingsystem == "windows"
     vardir = File.join(Dir::COMMON_APPDATA, "PuppetLabs", "puppet", "cache")
   else
-    vardir = "/opt/puppetlabs/puppet/cache"
+    vardir = "/var/lib/puppet"
   end
 
   if not InstallOptions.rundir.nil?
@@ -304,7 +304,7 @@ def prepare_installation
     if $operatingsystem == "windows"
       localedir = File.join(Dir::PROGRAM_FILES, "Puppet Labs", "Puppet", "puppet", "share", "locale")
     else
-      localedir = "/opt/puppetlabs/puppet/share/locale"
+      localedir = "/usr/share/puppetlabs/puppet/locale"
     end
   end

diff --git a/lib/puppet/defaults.rb b/lib/puppet/defaults.rb
index 39834ff..2132888 100644
--- a/lib/puppet/defaults.rb
+++ b/lib/puppet/defaults.rb
@@ -41,7 +41,7 @@ module Puppet
       end
       path.join(File::PATH_SEPARATOR)
     else
-      '$codedir/modules:/opt/puppetlabs/puppet/modules'
+      '$codedir/modules:/usr/share/puppetlabs/puppet/modules'
     end
   end

@@ -54,7 +54,7 @@ module Puppet
         nil
       end
     else
-      '/opt/puppetlabs/puppet/vendor_modules'
+      '/usr/share/puppetlabs/puppet/vendor_modules'
     end
   end

diff --git a/lib/puppet/face/config.rb b/lib/puppet/face/config.rb
index 692d12b..2d98e8c 100644
--- a/lib/puppet/face/config.rb
+++ b/lib/puppet/face/config.rb
@@ -131,7 +131,7 @@ Puppet::Face.define(:config, '0.0.1') do

       Set the vardir for only the agent:

-      $ puppet config set vardir /opt/puppetlabs/puppet/cache --section agent
+      $ puppet config set vardir /var/lib/puppet --section agent
     EOT

     when_invoked do |name, value, options|
diff --git a/lib/puppet/face/module/install.rb b/lib/puppet/face/module/install.rb
index 62068a5..518b0cc 100644
--- a/lib/puppet/face/module/install.rb
+++ b/lib/puppet/face/module/install.rb
@@ -47,20 +47,20 @@ Puppet::Face.define(:module, '1.0.0') do

       Install a module into a specific directory:

-      $ puppet module install puppetlabs-vcsrepo --target-dir=/opt/puppetlabs/puppet/modules
-      Preparing to install into /opt/puppetlabs/puppet/modules ...
+      $ puppet module install puppetlabs-vcsrepo --target-dir=/usr/share/puppetlabs/puppet/modules
+      Preparing to install into /usr/share/puppetlabs/puppet/modules ...
       Downloading from https://forgeapi.puppet.com ...
       Installing -- do not interrupt ...
-      /opt/puppetlabs/puppet/modules
+      /usr/share/puppetlabs/puppet/modules
       └── puppetlabs-vcsrepo (v0.0.4)

       Install a module into a specific directory and check for dependencies in other directories:

-      $ puppet module install puppetlabs-vcsrepo --target-dir=/opt/puppetlabs/puppet/modules --modulepath /etc/puppetlabs/code/modules
-      Preparing to install into /opt/puppetlabs/puppet/modules ...
+      $ puppet module install puppetlabs-vcsrepo --target-dir=/usr/share/puppetlabs/puppet/modules --modulepath /etc/puppetlabs/code/modules
+      Preparing to install into /usr/share/puppetlabs/puppet/modules ...
       Downloading from https://forgeapi.puppet.com ...
       Installing -- do not interrupt ...
-      /opt/puppetlabs/puppet/modules
+      /usr/share/puppetlabs/puppet/modules
       └── puppetlabs-vcsrepo (v0.0.4)

       Install a module from a release archive:
diff --git a/lib/puppet/face/module/list.rb b/lib/puppet/face/module/list.rb
index 001a736..946dcfe 100644
--- a/lib/puppet/face/module/list.rb
+++ b/lib/puppet/face/module/list.rb
@@ -27,7 +27,7 @@ Puppet::Face.define(:module, '1.0.0') do
         ├── puppetlabs-mysql (v0.0.1)
         ├── puppetlabs-sqlite (v0.0.1)
         └── puppetlabs-stdlib (v2.2.1)
-        /opt/puppetlabs/puppet/modules (no modules installed)
+        /usr/share/puppetlabs/puppet/modules (no modules installed)

       List installed modules in a tree view:

@@ -38,7 +38,7 @@ Puppet::Face.define(:module, '1.0.0') do
           ├─┬ puppetlabs-mysql (v0.0.1)
           │ └── bodepd-create_resources (v0.0.1)
           └── puppetlabs-sqlite (v0.0.1)
-        /opt/puppetlabs/puppet/modules (no modules installed)
+        /usr/share/puppetlabs/puppet/modules (no modules installed)

       List installed modules from a specified environment:

@@ -49,12 +49,12 @@ Puppet::Face.define(:module, '1.0.0') do
         ├── puppetlabs-mysql (v0.0.1)
         ├── puppetlabs-sqlite (v0.0.1)
         └── puppetlabs-stdlib (v2.2.1)
-        /opt/puppetlabs/puppet/modules (no modules installed)
+        /usr/share/puppetlabs/puppet/modules (no modules installed)

       List installed modules from a specified modulepath:

-      $ puppet module list --modulepath /opt/puppetlabs/puppet/modules
-        /opt/puppetlabs/puppet/modules (no modules installed)
+      $ puppet module list --modulepath /usr/share/puppetlabs/puppet/modules
+        /usr/share/puppetlabs/puppet/modules (no modules installed)
     EOT

     when_invoked do |options|
diff --git a/lib/puppet/face/module/uninstall.rb b/lib/puppet/face/module/uninstall.rb
index 6e73bc6..7fa5ec7 100644
--- a/lib/puppet/face/module/uninstall.rb
+++ b/lib/puppet/face/module/uninstall.rb
@@ -17,8 +17,8 @@ Puppet::Face.define(:module, '1.0.0') do

       Uninstall a module from a specific directory:

-      $ puppet module uninstall puppetlabs-ssh --modulepath /opt/puppetlabs/puppet/modules
-      Removed /opt/puppetlabs/puppet/modules/ssh (v1.0.0)
+      $ puppet module uninstall puppetlabs-ssh --modulepath /usr/share/puppetlabs/puppet/modules
+      Removed /usr/share/puppetlabs/puppet/modules/ssh (v1.0.0)

       Uninstall a module from a specific environment:

diff --git a/lib/puppet/provider/package/puppet_gem.rb b/lib/puppet/provider/package/puppet_gem.rb
index e5cd381..64cb6f5 100644
--- a/lib/puppet/provider/package/puppet_gem.rb
+++ b/lib/puppet/provider/package/puppet_gem.rb
@@ -12,6 +12,6 @@ Puppet::Type.type(:package).provide :puppet_gem, :parent => :gem do
     # sort out the absolute path.
     commands :gemcmd => "gem"
   else
-    commands :gemcmd => "/opt/puppetlabs/puppet/bin/gem"
+    commands :gemcmd => "/usr/bin/gem"
   end
 end
diff --git a/lib/puppet/reference/configuration.rb b/lib/puppet/reference/configuration.rb
index e747f9a..40882d8 100644
--- a/lib/puppet/reference/configuration.rb
+++ b/lib/puppet/reference/configuration.rb
@@ -23,7 +23,7 @@ config = Puppet::Util::Reference.newreference(:configuration, :depth => 1, :doc
     # Now print the data about the item.
     val = object.default
     if name.to_s == 'vardir'
-      val = 'Unix/Linux: /opt/puppetlabs/puppet/cache -- Windows: C:\ProgramData\PuppetLabs\puppet\cache -- Non-root user: ~/.puppetlabs/opt/puppet/cache'
+      val = 'Unix/Linux: /var/lib/puppet -- Windows: C:\ProgramData\PuppetLabs\puppet\cache -- Non-root user: ~/.puppetlabs/opt/puppet/cache'
     elsif name.to_s == 'confdir'
       val = 'Unix/Linux: /etc/puppetlabs/puppet -- Windows: C:\ProgramData\PuppetLabs\puppet\etc -- Non-root user: ~/.puppetlabs/etc/puppet'
     elsif name.to_s == 'codedir'
diff --git a/lib/puppet/util/run_mode.rb b/lib/puppet/util/run_mode.rb
index d0f9b9a..e7ecd40 100644
--- a/lib/puppet/util/run_mode.rb
+++ b/lib/puppet/util/run_mode.rb
@@ -67,7 +67,7 @@ module Puppet
       end

       def var_dir
-        which_dir("/opt/puppetlabs/puppet/cache", "~/.puppetlabs/opt/puppet/cache")
+        which_dir("/var/lib/puppet", "~/.puppetlabs/opt/puppet/cache")
       end

       def run_dir
diff --git a/man/man5/puppet.conf.5 b/man/man5/puppet.conf.5
index 8b46f57..5905fde 100644
--- a/man/man5/puppet.conf.5
+++ b/man/man5/puppet.conf.5
@@ -111,7 +111,7 @@ The search path for \fBglobal\fR modules\. Should be specified as a list of dire
 These are the modules that will be used by \fIall\fR environments\. Note that the \fBmodules\fR directory of the active environment will have priority over any global directories\. For more info, see \fIhttps://puppet\.com/docs/puppet/latest/environments_about\.html\fR
 .
 .IP "\(bu" 4
-\fIDefault\fR: $codedir/modules:/opt/puppetlabs/puppet/modules
+\fIDefault\fR: $codedir/modules:/usr/share/puppetlabs/puppet/modules
 .
 .IP "" 0
 .
@@ -1931,7 +1931,7 @@ The user Puppet Server will run as\. Used to ensure the agent side processes (ag
 Where Puppet stores dynamic and growing data\. The default for this setting is calculated specially, like \fBconfdir\fR_\.
 .
 .IP "\(bu" 4
-\fIDefault\fR: Unix/Linux: /opt/puppetlabs/puppet/cache \-\- Windows: C:\eProgramData\ePuppetLabs\epuppet\ecache \-\- Non\-root user: ~/\.puppetlabs/opt/puppet/cache
+\fIDefault\fR: Unix/Linux: /var/lib/puppet \-\- Windows: C:\eProgramData\ePuppetLabs\epuppet\ecache \-\- Non\-root user: ~/\.puppetlabs/opt/puppet/cache
 .
 .IP "" 0
 .
@@ -1939,7 +1939,7 @@ Where Puppet stores dynamic and growing data\. The default for this setting is c
 The directory containing \fBvendored\fR modules\. These modules will be used by \fIall\fR environments like those in the \fBbasemodulepath\fR\. The only difference is that modules in the \fBbasemodulepath\fR are pluginsynced, while vendored modules are not
 .
 .IP "\(bu" 4
-\fIDefault\fR: /opt/puppetlabs/puppet/vendor_modules
+\fIDefault\fR: /usr/share/puppetlabs/puppet/vendor_modules
 .
 .IP "" 0
 .
diff --git a/man/man8/puppet-config.8 b/man/man8/puppet-config.8
index 3c8e65d..11db793 100644
--- a/man/man8/puppet-config.8
+++ b/man/man8/puppet-config.8
@@ -145,7 +145,7 @@ $ puppet config set rundir /var/run/puppetlabs
 Set the vardir for only the agent:
 .
 .P
-$ puppet config set vardir /opt/puppetlabs/puppet/cache \-\-section agent
+$ puppet config set vardir /var/lib/puppet \-\-section agent
 .
 .SH "COPYRIGHT AND LICENSE"
 Copyright 2011 by Puppet Inc\. Apache 2 license; see COPYING
diff --git a/man/man8/puppet-module.8 b/man/man8/puppet-module.8
index 541142c..1f3ce39 100644
--- a/man/man8/puppet-module.8
+++ b/man/man8/puppet-module.8
@@ -271,13 +271,13 @@ $ puppet module install puppetlabs\-vcsrepo \-v 0\.0\.4 Preparing to install int
 Install a module into a specific directory:
 .
 .P
-$ puppet module install puppetlabs\-vcsrepo \-\-target\-dir=/opt/puppetlabs/puppet/modules Preparing to install into /opt/puppetlabs/puppet/modules \.\.\. Downloading from https://forgeapi\.puppet\.com \.\.\. Installing \-\- do not interrupt \.\.\. /opt/puppetlabs/puppet/modules └── puppetlabs\-vcsrepo (v0\.0\.4)
+$ puppet module install puppetlabs\-vcsrepo \-\-target\-dir=/usr/share/puppetlabs/puppet/modules Preparing to install into /usr/share/puppetlabs/puppet/modules \.\.\. Downloading from https://forgeapi\.puppet\.com \.\.\. Installing \-\- do not interrupt \.\.\. /usr/share/puppetlabs/puppet/modules └── puppetlabs\-vcsrepo (v0\.0\.4)
 .
 .P
 Install a module into a specific directory and check for dependencies in other directories:
 .
 .P
-$ puppet module install puppetlabs\-vcsrepo \-\-target\-dir=/opt/puppetlabs/puppet/modules \-\-modulepath /etc/puppetlabs/code/modules Preparing to install into /opt/puppetlabs/puppet/modules \.\.\. Downloading from https://forgeapi\.puppet\.com \.\.\. Installing \-\- do not interrupt \.\.\. /opt/puppetlabs/puppet/modules └── puppetlabs\-vcsrepo (v0\.0\.4)
+$ puppet module install puppetlabs\-vcsrepo \-\-target\-dir=/usr/share/puppetlabs/puppet/modules \-\-modulepath /etc/puppetlabs/code/modules Preparing to install into /usr/share/puppetlabs/puppet/modules \.\.\. Downloading from https://forgeapi\.puppet\.com \.\.\. Installing \-\- do not interrupt \.\.\. /usr/share/puppetlabs/puppet/modules └── puppetlabs\-vcsrepo (v0\.0\.4)
 .
 .P
 Install a module from a release archive:
@@ -298,25 +298,25 @@ $ puppet module install puppetlabs\-vcsrepo\-0\.0\.4\.tar\.gz \-\-ignore\-depend
 List installed modules:
 .
 .P
-$ puppet module list /etc/puppetlabs/code/modules ├── bodepd\-create_resources (v0\.0\.1) ├── puppetlabs\-bacula (v0\.0\.2) ├── puppetlabs\-mysql (v0\.0\.1) ├── puppetlabs\-sqlite (v0\.0\.1) └── puppetlabs\-stdlib (v2\.2\.1) /opt/puppetlabs/puppet/modules (no modules installed)
+$ puppet module list /etc/puppetlabs/code/modules ├── bodepd\-create_resources (v0\.0\.1) ├── puppetlabs\-bacula (v0\.0\.2) ├── puppetlabs\-mysql (v0\.0\.1) ├── puppetlabs\-sqlite (v0\.0\.1) └── puppetlabs\-stdlib (v2\.2\.1) /usr/share/puppetlabs/puppet/modules (no modules installed)
 .
 .P
 List installed modules in a tree view:
 .
 .P
-$ puppet module list \-\-tree /etc/puppetlabs/code/modules └─┬ puppetlabs\-bacula (v0\.0\.2) ├── puppetlabs\-stdlib (v2\.2\.1) ├─┬ puppetlabs\-mysql (v0\.0\.1) │ └── bodepd\-create_resources (v0\.0\.1) └── puppetlabs\-sqlite (v0\.0\.1) /opt/puppetlabs/puppet/modules (no modules installed)
+$ puppet module list \-\-tree /etc/puppetlabs/code/modules └─┬ puppetlabs\-bacula (v0\.0\.2) ├── puppetlabs\-stdlib (v2\.2\.1) ├─┬ puppetlabs\-mysql (v0\.0\.1) │ └── bodepd\-create_resources (v0\.0\.1) └── puppetlabs\-sqlite (v0\.0\.1) /usr/share/puppetlabs/puppet/modules (no modules installed)
 .
 .P
 List installed modules from a specified environment:
 .
 .P
-$ puppet module list \-\-environment production /etc/puppetlabs/code/modules ├── bodepd\-create_resources (v0\.0\.1) ├── puppetlabs\-bacula (v0\.0\.2) ├── puppetlabs\-mysql (v0\.0\.1) ├── puppetlabs\-sqlite (v0\.0\.1) └── puppetlabs\-stdlib (v2\.2\.1) /opt/puppetlabs/puppet/modules (no modules installed)
+$ puppet module list \-\-environment production /etc/puppetlabs/code/modules ├── bodepd\-create_resources (v0\.0\.1) ├── puppetlabs\-bacula (v0\.0\.2) ├── puppetlabs\-mysql (v0\.0\.1) ├── puppetlabs\-sqlite (v0\.0\.1) └── puppetlabs\-stdlib (v2\.2\.1) /usr/share/puppetlabs/puppet/modules (no modules installed)
 .
 .P
 List installed modules from a specified modulepath:
 .
 .P
-$ puppet module list \-\-modulepath /opt/puppetlabs/puppet/modules /opt/puppetlabs/puppet/modules (no modules installed)
+$ puppet module list \-\-modulepath /usr/share/puppetlabs/puppet/modules /usr/share/puppetlabs/puppet/modules (no modules installed)
 .
 .P
 \fBsearch\fR
@@ -340,7 +340,7 @@ $ puppet module uninstall puppetlabs\-ssh Removed /etc/puppetlabs/code/modules/s
 Uninstall a module from a specific directory:
 .
 .P
-$ puppet module uninstall puppetlabs\-ssh \-\-modulepath /opt/puppetlabs/puppet/modules Removed /opt/puppetlabs/puppet/modules/ssh (v1\.0\.0)
+$ puppet module uninstall puppetlabs\-ssh \-\-modulepath /usr/share/puppetlabs/puppet/modules Removed /usr/share/puppetlabs/puppet/modules/ssh (v1\.0\.0)
 .
 .P
 Uninstall a module from a specific environment:
diff --git a/spec/unit/defaults_spec.rb b/spec/unit/defaults_spec.rb
index 9a6dc40..0632839 100644
--- a/spec/unit/defaults_spec.rb
+++ b/spec/unit/defaults_spec.rb
@@ -132,7 +132,7 @@ describe "Defaults" do
     it 'includes the user and system modules', :unless => Puppet::Util::Platform.windows? do
       expect(
         Puppet[:basemodulepath]
-      ).to match(%r{.*/code/modules:/opt/puppetlabs/puppet/modules$})
+      ).to match(%r{.*/code/modules:/usr/share/puppetlabs/puppet/modules$})
     end
 
     describe 'on windows', :if => Puppet::Util::Platform.windows? do
@@ -160,7 +160,7 @@ describe "Defaults" do
     it 'includes the default vendormoduledir', :unless => Puppet::Util::Platform.windows? do
       expect(
         Puppet[:vendormoduledir]
-      ).to eq('/opt/puppetlabs/puppet/vendor_modules')
+      ).to eq('/usr/share/puppetlabs/puppet/vendor_modules')
     end
 
     describe 'on windows', :if => Puppet::Util::Platform.windows? do
diff --git a/spec/unit/provider/package/puppet_gem_spec.rb b/spec/unit/provider/package/puppet_gem_spec.rb
index 07e5fba..760af61 100644
--- a/spec/unit/provider/package/puppet_gem_spec.rb
+++ b/spec/unit/provider/package/puppet_gem_spec.rb
@@ -17,7 +17,7 @@ describe Puppet::Type.type(:package).provider(:puppet_gem) do
   if Puppet::Util::Platform.windows?
     let(:provider_gem_cmd) { 'gem' }
   else
-    let(:provider_gem_cmd) { '/opt/puppetlabs/puppet/bin/gem' }
+    let(:provider_gem_cmd) { '/usr/bin/gem' }
   end

   let(:execute_options) { {:failonfail => true, :combine => true, :custom_environment => {"HOME"=>ENV["HOME"]}} }
diff --git a/spec/unit/util/run_mode_spec.rb b/spec/unit/util/run_mode_spec.rb
index 81855b1..d619588 100644
--- a/spec/unit/util/run_mode_spec.rb
+++ b/spec/unit/util/run_mode_spec.rb
@@ -73,8 +73,8 @@ describe Puppet::Util::RunMode do
     end

     describe "#var_dir" do
-      it "has vardir /opt/puppetlabs/puppet/cache when run as root" do
-        as_root { expect(@run_mode.var_dir).to eq(File.expand_path('/opt/puppetlabs/puppet/cache')) }
+      it "has vardir /usr/share/puppetlabs/puppet/cache when run as root" do
+        as_root { expect(@run_mode.var_dir).to eq(File.expand_path('/usr/share/puppetlabs/puppet/cache')) }
       end

       it "has vardir ~/.puppetlabs/opt/puppet/cache when run as non-root" do
