diff --git a/lib/puppet/configurer.rb b/lib/puppet/configurer.rb
index 2ccddbe..4c6e33d 100644
--- a/lib/puppet/configurer.rb
+++ b/lib/puppet/configurer.rb
@@ -1,5 +1,4 @@
 # The client for interacting with the puppetmaster config server.
-require 'sync'
 require 'timeout'
 require 'puppet/network/http_pool'
 require 'puppet/util'
diff --git a/lib/puppet/util/storage.rb b/lib/puppet/util/storage.rb
index 713f650..b9183ca 100644
--- a/lib/puppet/util/storage.rb
+++ b/lib/puppet/util/storage.rb
@@ -1,5 +1,4 @@
 require 'yaml'
-require 'sync'
 require 'singleton'
 require 'puppet/util/yaml'
 
